test/integration/helpers_test.go

```diff
@@ -0,0 +1,173 @@
+}
+
+// gatewayHealthCheck checks the gateway has been scheduled by KIC. This function is inspired by
+// assert.eventually (https://github.com/stretchr/testify/blob/v1.7.5/assert/assertions.go#L1669-L1700)
+// and patched with custom behavior to determine the health of the gateway and to return an error
+// instead of failing (at the time of its call, we don't have any test instance to make fail yet).
+func gatewayHealthCheck(ctx context.Context, client *gatewayclient.Clientset, gatewayName, namespace string) error {
+	ch := make(chan bool, 1)
+
+	timer := time.NewTimer(gatewayWaitTimeToVerifyScheduling)
+	defer timer.Stop()
+
+	ticker := time.NewTicker(time.Millisecond * 200)
+	defer ticker.Stop()
+
+	for tick := ticker.C; ; {
+		select {
+		case <-timer.C:
+			return fmt.Errorf("the gateway %s/%s did not become scheduled after %s", namespace, defaultGatewayName, gatewayWaitTimeToVerifyScheduling)
+		case <-tick:
+			tick = nil
+			ch <- func() bool {
+				gw, err := client.GatewayV1alpha2().Gateways(namespace).Get(ctx, gatewayName, metav1.GetOptions{})
+				exitOnErr(err)
+				for _, cond := range gw.Status.Conditions {
+					if cond.Reason == string(gatewayv1alpha2.GatewayReasonReady) {
+						return true
+					}
+				}
+				return false
+			}()
+		case v := <-ch:
+			if v {
+				return nil
+			}
+			tick = ticker.C
+		}
+	}
+}
+
+// eventuallyGatewayLinkStatusMatches checks if the specific Route (HTTP, TCP, TLS, or UDP)
+// is correctly linked to a supported gateway
+func GetGatewayIsLinkedCallback(t *testing.T, c *gatewayclient.Clientset, protocolType gatewayv1alpha2.ProtocolType, namespace, name string) func() bool {
+	return func() bool {
+		return gatewayLinkStatusMatches(t, c, true, protocolType, namespace, name)
+	}
+}
+
+// eventuallyGatewayLinkStatusMatches checks if the specific Route (HTTP, TCP, TLS, or UDP)
+// is correctly unlinked from a supported gateway
+func GetGatewayIsUnlinkedCallback(t *testing.T, c *gatewayclient.Clientset, protocolType gatewayv1alpha2.ProtocolType, namespace, name string) func() bool {
+	return func() bool {
+		return gatewayLinkStatusMatches(t, c, false, protocolType, namespace, name)
+	}
+}
+
+// gatewayLinkStatusMatches checks if the specific Route (HTTP, TCP, TLS, or UDP)
+// is correctly linked to (or unlinked from) a supported gateway. In order to assert
+// that the route must be linked to the gateway, or unlinked from the gateway, the
+// verifyLinked boolean arg must be set accordingly.
+func gatewayLinkStatusMatches(t *testing.T, c *gatewayclient.Clientset, verifyLinked bool, protocolType gatewayv1alpha2.ProtocolType, namespace, name string) bool {
+	var routeParents []gatewayv1alpha2.RouteParentStatus
+
+	// gather a fresh copy of the route, given the specific protocol type
+	switch protocolType {
+	case gatewayv1alpha2.HTTPProtocolType:
+		route, err := c.GatewayV1alpha2().HTTPRoutes(namespace).Get(ctx, name, metav1.GetOptions{})
+		t.Logf("error %v in getting http route", err)
```

Comment:
Personal preference maybe but I'd use the following formatting for increased (arguable) readability:

```suggestion
		t.Logf("error getting http route: %v", err)
```